server/routes/dexcom.js

```diff
@@ -1,14 +1,53 @@
 const fetch = require('node-fetch');
+const { refreshDexcomToken } = require('../auth/strategies/dexcom');
 
-function dexReq(req) {
-  fetch(`https://sandbox-api.dexcom.com/v2/users/self/egvs?startDate=${req.query.start}&endDate=${req.query.end}`, {
-          method: 'get',
-          headers: {
-            'Authorization': 'Bearer ' + req.user.dexcomAccessToken
-          },
-      })
-      .then(res => res.json())
-      .then(json => console.log(json));
+async function dexReq(req) {
+  try {
+    var response = await tryFetch(req);
+    if (response === 401) {
+      await refreshDexcomToken(req.user);
+      response = await tryFetch(req);
+    }
+    if (response === 401) {
+      return { 'error': 'unauthorized' };
+    }
+    const json = await response.json();
+    return parseData(json);
+  } catch (error) {
+    console.log(error);
+    return {};
+  }
+}
+
+async function tryFetch(req) {
+  var response = await fetchDex(req);
+  var { status } = response;
+  if (status === 401) {
+    return status;
+  }
+  return response;
+}
+
+async function fetchDex(req) {
+  return fetch(`https://sandbox-api.dexcom.com/v2/users/self/egvs?startDate=${req.query.start}&endDate=${req.query.end}`, {
+    method: 'get',
+    headers: {
+      'Authorization': 'Bearer ' + req.user.dexcomAccessToken
+    },
+  })
+}
+
+function parseData(json) {
+  var ret = { 'data': [] };
+
+  for(let val of json.egvs) {
```

Comment:
it'd be better to use [map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) here